docs: clarify newsletter useFetcher example in the docs (#1482)
remix-run/remix: docs/api/remix.md

**Newsletter Signup Form**

> Perhaps you have a persistent newsletter signup at the bottom of every page on your site. This is not a navigation event, so useFetcher is perfect for the job. First, you create an API-only route:

Reviewer: ```suggestion
Perhaps you have a persistent newsletter signup at the bottom of every page on your site. This is not a navigation event, so useFetcher is perfect for the job. First, you create a Resource Route:
```
